### ISSUE TITLE
F03: parameterized derived types

### ISSUE THREAD

**Issue by shivaramaarao** [flang-compiler/flang#866]
when a basic program with f03 parameterized derived type is tried with flang it doesnt work. Is this feature unsupported in flang or is it broken ?

Program test
Type myt(k)
   Integer,Len::k
   Real  a(k)
End Type
  Type(myt(1)) z
  z%a(7) = 10
  print *, z%a
End Program

flang k1.f90
./a.out
Segmentation fault (core dumped)

**Comment by mleair** [comment]
z%a(7) =10 is referencing outside its bounds since 1 was specified for length parameter k. Saying that, we still have a bug in the declaration of z. Adding colons to the declaration appears to work. For example,

Type(myt(1)) :: z


And as noted above, you'll need to change the assignment to:

z%a(1) = 10


Of course, if you really want z%a to have 7 elements, then change your declaration to:

Type(myt(7)) :: z

(if you want z%a to have more than 7 elements, then declare it as such)

In summary, the full program below works with Flang.

Program test
Type myt(k)
Integer,Len::k
Real :: a(k)
End Type
Type(myt(7)) :: z
z%a(7) = 10
print *, z%a
End Program

We do have a bug, but it's in the declaration form that does not include colons (i.e., Type(myt(1) z). As a work around, use colons (i.e., Tyoe(myt(1) :: z).

**Comment by shivaramaarao** [comment]
thanks.   I made a mistake in uploading the test with Type(myt(1)) z. it should have been Type(myt(7)) z.

As you mentioned the issue is that colon is not included. when colon is added the testcase works fine.

Thanks
